### PR TITLE
Singlepulse 

### PR DESCRIPTION
**Pull request by chitrangpatel**
Hi Scott, can you please merge this pull request. 

It now contains automatic waterfalling, spd file generation and all the other new single pulse pipeline additions. This is not specific to PALFA anymore and can be used by other surveys as well. 

Paul and I have tested this version of presto on a bunch of candidates and it works fine.

### CODE REVIEW

**Comment by scottransom** [comment]
Wow.  Big merge.  I'll take a look.  One big red flag right away are the changes to the Makefile.  The current version works on many architectures, but your changes would break even that.  Please remove the Makefile stuff.  More later...

**Comment by chitrangpatel** [comment]
I fixed the makefile problem. Awaiting further instructions.

**Comment by scottransom** [comment]
One more easy request:  add anyone who is applicable for this (excellent!) contribution to the README file as a contributor!  :-)

File: README.md [hunk 1]
```diff
@@ -154,7 +157,7 @@ code, Jason Hessels for many contributions to the Python routines, and
 a bunch of other contributions of various kinds from (alphabetical):
 Anne Archibald, Cees Bassa, Slavko Bogdanov, Fernando Camilo, Paul
 Demorest, Paulo Freire, Mike Keith, Patrick Lazarus, Maggie
-Livingstone, Paul Ray, Paul Scholz, Ingrid Stairs, Kevin Stovall, and
-for many comments, suggestions and patches!
+Livingstone, Paul Ray, Paul Scholz, Ingrid Stairs, Kevin Stovall, Chen
```
Comment (scottransom) [inline README.md]: Heh.  This list of contributors is alphabetical!  :-)

**Comment by scottransom** [comment]
Whoa...  I just realized that I had forgotten about this.  Is it still all ready to go?

**Comment by chitrangpatel** [comment]
Unless there are any red flags from your end, it is good to go.

**Pull request merged**